### PR TITLE
fix: Various NavMobile and NavJumpMenu fixes for EDU

### PR DESCRIPTION
**Pull request by stephiescastle**
### Checklist

- [x] Include a description of your pull request and instructions for the reviewer to verify your work.
- [x] Link to the issue if this PR is issue-specific.
- [ ] Create/update the corresponding story if this includes a UI component.
- [ ] Create/update documentation. If not included, tell us why.
- [x] List the environments / browsers in which you tested your changes.
- [x] Tests, linting, or other required checks are passing.
- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../releases); good ones make that easier to scan.
  - PRs will be broadly categorized in the change log, but for even easier scanning, consider prefixing with a component name or other useful categorization, e.g., "BaseButton: fix layout bug", or "Storybook: Update dependencies".
- [x] PR has been tagged with a [SemVer](https://semver.org/) label and a [general category label](https://github.com/nasa-jpl/explorer-1/blob/52994b671411f55961d7beb8851d4580ac3f434f/.github/release-drafter.config.yml#L21-L39), or skip-changelog.
  - These tags are used to do the aforementioned broad categorization in the change log and determine what the next release's version number should be.
  - [Release Drafter](https://github.com/marketplace/actions/release-drafter) will attempt to do the category labeling for you! Please double-check its work.

## Description

Fixes regressions (some introduced in https://github.com/nasa-jpl/explorer-1/pull/656/):
- https://github.com/nasa-jpl/www/issues/700
- https://github.com/nasa-jpl/www/issues/702
- https://github.com/nasa-jpl/www/issues/704

Changes:
- sets `overflow-x` to hidden on "Section" divs within the EDU Lesson template and EDU Student Project templates
- adds back `hidden` class to NavJumpMenu at smaller screen sizes
- wraps `Education` in `BaseLink` in `NavMobile`

### Operating System

- [x] macOS
- [ ] iOS
- [ ] iPadOS
- [ ] Windows

### Browser

- [x] Chrome
- [x] Firefox ESR
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by jamesray — APPROVED** [review]

File: packages/vue/src/components/NavMobile/NavMobile.vue [hunk 1]
```diff
@@ -41,7 +41,13 @@
               v-if="themeStore.theme === 'ThemeEdu'"
               class="-ml-1 pl-2 border-l border-white border-opacity-30 z-20 print:border-black"
             >
-              <span class="text-white font-bold text-xl pl-px">Education</span>
+              <BaseLink
+                class="text-white font-bold text-xl pl-px"
+                to="/edubeta/"
```
Comment (Scotchester) [inline packages/vue/src/components/NavMobile/NavMobile.vue:46]: Note to update #660 for this after it's merged.

**Review by Scotchester — COMMENTED** [review]
The horizontal overflow is fixed, but I'm still not seeing the on-page jump menu.

**Comment by stephiescastle** [comment]
> The horizontal overflow is fixed, but I'm still not seeing the on-page jump menu.

It's not intended to be visible on mobile. It's part of the mobile navigation. Would be nice to have it be visible but decided to wait on that as it's a bit more involved and not original spec.

**Comment by Scotchester** [comment]
> > The horizontal overflow is fixed, but I'm still not seeing the on-page jump menu.
> 
> It's not intended to be visible on mobile. It's part of the mobile navigation. Would be nice to have it be visible but decided to wait on that as it's a bit more involved and not original spec.

Ahhh, thanks for clarifying.

**Pull request merged**